frontend/src/components/cluster-dashboard/cluster-list-view/ClusterListView.tsx

```diff
@@ -0,0 +1,158 @@
+
+type ClusterListViewProps = {
+  clusterData: Cluster;
+  refreshClusterData: () => void;
+}
+
+const ClusterListView: React.FC<ClusterListViewProps> = (props: ClusterListViewProps) => {
+
+  const [openSnackBar, setOpenSnackBar] = React.useState(false);
+  const [snackBarMessage, setSnackBarMessage] = React.useState("");
+  const [openConfDialog, setOpenConfDialog] = React.useState(false);
+  const [confDialogTitle, setConfDialogTitle] = React.useState("");
+  const [confDialogContent, setConfDialogContent] = React.useState("");
+  const [confDialogData, setConfDialogData] = React.useState({});
+  const [addNodeAddress, setAddNodeAddress] = React.useState("");
+  const clusterData: Cluster = props.clusterData;
+  const akkaGetClusterMembersUrl = akkaClusterProps["akka.management.url"] + "/cluster/members";
+
+  const handleMember = (member: string, mode: string): void => {
+    setConfDialogTitle(mode.toLowerCase() === "leave" ? "Remove Node From Cluster?" : "Shutdown Node in Cluster?");
+    setConfDialogContent(
+      mode.toLowerCase() === "leave" ? 
+      "Are you sure you want to remove node " + member + " from the cluster?" : 
+      "Are you sure you want to shutdown node " + member + " in the cluster?"
+    );
+    setConfDialogData({ member: member, mode: mode });
+    setOpenConfDialog(true);
+  }
+
+  const addMember = (): void => {
+    if(!addNodeAddress || addNodeAddress.trim() === "") { return; }
+    
+    axios.post(akkaGetClusterMembersUrl, 
+        querystring.stringify({ address: addNodeAddress.trim() }),
+        { headers: { "Content-Type": "application/x-www-form-urlencoded" }}
+      )
+      .then(response => {
+        if(response.status === 200) {
+          setSnackBarMessage(response.data.message);
+          setOpenSnackBar(true);
+          props.refreshClusterData();
+        }
+      })
+      .catch(function (error) {
+        console.log(error);
+        setSnackBarMessage(error.message && error.message.message ? error.message.message : error.message);
+        setOpenSnackBar(true);
+      })
+  }
+
+  const handleMemberConfirm = ({ member, mode }: { member: string; mode: string }): void => {
+    setOpenConfDialog(false);
+    axios.put(akkaGetClusterMembersUrl + "/" + member.split('://')[1], 
+        querystring.stringify({ operation: mode }),
+        { headers: { "Content-Type": "application/x-www-form-urlencoded" }}
+      )
+      .then(response => {
+        if(response.status === 200) {
+          setSnackBarMessage(response.data.message);
+          setOpenSnackBar(true);
+          props.refreshClusterData();
+        }
+      })
+      .catch(function (error) {
+        console.log(error);
+        setSnackBarMessage(error.message && error.message.message ? error.message.message : error.message);
+        setOpenSnackBar(true);
+      })
+  }
+
+  return (
+    <React.Fragment>
+      <Grid item xs={3} className="shukra-left-container pos-rel">
```

Comment:
This component is acting a container for the whole view, it seems, so maybe it would be possible to split this grid into its own component, to somewhat reduce the complexity of this file.